templates/solaris_install.sh.erb

```diff
@@ -1,11 +1,13 @@
 function start_service() {
   service="${1:?}"
-  /opt/puppetlabs/bin/puppet resource service "${service:?}" ensure=running enable=true
+  /opt/puppetlabs/bin/puppet resource service "${service:?}" ensure=running
 }
-
+service_was_running_before_agent_update = false
+if /opt/puppetlabs/bin/puppet resource service "${service_name:?}" | grep "ensure => 'running'" ; then
```

Comment:
Where does this service_name variable come from? I think you're confusing shell syntax with erb templating. This is definitely getting a little complicated.

I think this would work
```
<% @service_names.each do |service_name| %>
  <%= service_name >_was_running = false
  if /opt/puppetlabs/bin/puppet resource service "<%= service_name>" | grep "ensure => 'running'" ; then
    <%= service_name >_was_running = true
  fi
<% end %>
```

then below
```
<% @service_names.each do |service_name| %>
  if $<%= service_name %>_was_running ; then
    start_service <%= service_name %>
  fi
<% end %>
```

Reply:
I think we want to update the spec tests to have some positive validation of what's produced too.

Reply:
I can handle the spec test bits. Working on what that would look like.

Reply:
Maybe something like https://github.com/puppetlabs/puppetlabs-puppet_agent/compare/master...MikaelSmith:MODULES-9173.